### PR TITLE
Add Remux Support

### PR DESCRIPTION
**Pull request by martinbowling**
Add Remux Quality Profile & Quality Parser

#### Database Migration
YES - Inserts new quality 11, uses open slot but should have a migration routine

#### Description
Create a new Quality Profile REMUX1080 and a Quality Parser to match these conditions. To shield people who don't understand what a REMUX is default profiles aren't touched, in my instance I created a new profile REMUX with a cut off of REMUX and a requirement of REMUX.

#### Todos
- [ ] Tests
- [ ] Documentation

#### Issues Fixed or Closed by this PR
#1774
*

### CODE REVIEW

**Comment by hthighway** [comment]
This would be a fantastic addition

File: src/NzbDrone.Core/Qualities/Quality.cs [hunk 1]
```diff
@@ -71,6 +71,8 @@ public override bool Equals(object obj)
         //public static Quality Bluray480p  { get { return new Quality(13, "Bluray-480p"); } }
         //public static Quality WEBRip720p  { get { return new Quality(14, "WEBRip-720p"); } }
         //public static Quality WEBRip1080p { get { return new Quality(15, "WEBRip-1080p"); } }
+        public static Quality REMUX1080p => new Quality(11, "REMUX-1080p");
```
Comment (markus101) [inline src/NzbDrone.Core/Qualities/Quality.cs]: `11` is already a place holder for `public static Quality HDTV480p    { get { return new Quality(11, "HDTV-480p"); } }`

File: src/NzbDrone.Core/Qualities/Quality.cs [hunk 2]
```diff
@@ -115,9 +118,10 @@ static Quality()
                 new QualityDefinition(Quality.Bluray720p)  { Weight = 9,  MinSize = 0, MaxSize = 100 },
                 new QualityDefinition(Quality.WEBDL1080p)  { Weight = 10, MinSize = 0, MaxSize = 100 },
                 new QualityDefinition(Quality.Bluray1080p) { Weight = 11, MinSize = 0, MaxSize = 100 },
-                new QualityDefinition(Quality.HDTV2160p)   { Weight = 12, MinSize = 0, MaxSize = null },
-                new QualityDefinition(Quality.WEBDL2160p)  { Weight = 13, MinSize = 0, MaxSize = null },
-                new QualityDefinition(Quality.Bluray2160p) { Weight = 14, MinSize = 0, MaxSize = null },
+                new QualityDefinition(Quality.REMUX1080p)   { Weight = 12, MinSize = 0, MaxSize = null },
```
Comment (markus101) [inline src/NzbDrone.Core/Qualities/Quality.cs]: Extraneous space.

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 3]
```diff
@@ -70,6 +73,16 @@ public static QualityModel ParseQuality(string name)
             var resolution = ParseResolution(normalizedName);
             var codecRegex = CodecRegex.Match(normalizedName);
 
+            if (RemuxRegex.IsMatch(normalizedName) && sourceMatch?.Groups["webdl"].Success != true && sourceMatch?.Groups["hdtv"].Success != true)
```
Comment (markus101) [inline src/NzbDrone.Core/Parser/QualityParser.cs]: There are a lot of other sources that could match here that should be excluded, specifying a bluray source would mean it'd match only when bluray and not have to exclude every other source that could be added.

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 4]
```diff
@@ -70,6 +73,16 @@ public static QualityModel ParseQuality(string name)
...
+            {
+
```
Comment (markus101) [inline src/NzbDrone.Core/Parser/QualityParser.cs]: Extraneous new line.

**Comment by Taloth** [comment]
@markus101 Remux quality is on the 3.x milestone, are we going to reverse our decision on that?

**Comment by markus101** [comment]
Good point @Taloth I don't think we want to reverse that, but it could go in v3.0,it's in v3.x for some flexibility to avoid piling on extra work for us, but if the work is done, no reason it can't go.

**Comment by Taloth** [comment]
Yes, I think we talked about it in the quality group stuff we already implemented on v3. Coz then you can put remux as part of bluray-1080p/2160p groups. I'd be ok in getting it into v3.0. Just gotta be careful with the quality id assignments and migrations.

**Comment by markus101** [comment]
Closing this as it hasn't progressed.

**Pull request closed without merging**